### PR TITLE
Si-strip commissioning workflow updates

### PR DESCRIPTION
**Pull request by rgerosa**
This PR includes:

a) Implementation of selective upload of channels to the online DB.
b) Removing invalid channels from the pedestals upload.
c) Re-design the analysis which fits for the si-strip pulse shape parameters for both deco and peak mode.
d) Update an old analysis task (PedsFullNoise) to be used to identify single noisy/bad strips in the detector. The analysis requires a new analysisDescription object provided by the external tkonline-libs. They have been already merged in CMSDIST via:
https://github.com/cms-sw/cmsdist/pull/3819

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by rgerosa** [comment]
I forgot to mention that this PR should not affect any standard workflow. The errors from the logs are  related to errors while opening files.

**Comment by perrotta** [comment]
Please @rgerosa have a look at the errors in the Unit tests

**Comment by mmusich** [comment]
@perrotta 
```
===== Test "testCalibTrackerSiStripCommon" ====
argument 0: testCalibTrackerSiStripCommon
argument 1: /bin/bash
argument 2: CalibTracker/SiStripCommon/test
argument 3: test_all.sh
shell is: /bin/bash
Current directory is: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300
topdir is: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300
testdir is: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300/src/CalibTracker/SiStripCommon/test
tmpdir is: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300/tmp/slc6_amd64_gcc630
testbin is: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300/test/slc6_amd64_gcc630
Running script: /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_X_2018-03-14-2300/src/CalibTracker/SiStripCommon/test/test_all.sh
TESTING CalibTracker/SiStripCommon ...
15-Mar-2018 16:12:45 CET  Initiating request to open file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_9_2_2/RelValTTbar_13/GEN-SIM-RECO/PU25ns_92X_upgrade2017_realistic_v1-v1/10000/ECFEA1BD-BF4D-E711-A404-0CC47A7C345C.root
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc


A fatal system signal has occurred: abort signal
The following is the call stack containing the origin of the signal.
```

look pretty much a transient issue related to this eos glitch: https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3462.html.
My suggestion is to re-trigger the tests.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Reco is involved with a few additions in a couple of DataFormats/SiStripCommon classes, whose usage is restricted to DQM and online monitoring
- No effect expected to be propagated to the reco quantities, none is seen
- jenkins tests pass

File: DQM/SiStripCommissioningClients/src/CalibrationHistograms.cc [hunk 1]
```diff
@@ -78,6 +81,10 @@ void CalibrationHistograms::histoAnalysis( bool debug ) {
       continue;
     }
     
+    std::cout.flush();
```
Comment (dmitrijus) [inline DQM/SiStripCommissioningClients/src/CalibrationHistograms.cc]: Please remove this printout.

**Comment by rgerosa** [comment]
@dmitrijus the branch has been updated accordingly

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Already signed by reco in https://github.com/cms-sw/cmssw/pull/22631#issuecomment-374416820, still fine after the last fix requested by @dmitrijus for two DQM modules
- jenkins tests still pass

**Comment by dmitrijus** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**